app/models/casa_case.rb

```diff
@@ -97,6 +98,30 @@ def has_transitioned?
     transition_aged_youth
   end
 
+  def contains_emancipation_option?(option_id)
+    begin
+      emancipation_options.find(option_id)
+    rescue ActiveRecord::RecordNotFound
+      return false
+    end
+
+    true
+  end
+
+  def add_emancipation_option(option_id)
+    option_category_id = EmancipationOption.find(option_id).emancipation_category_id
+
+    if !(EmancipationCategory.find(option_category_id).mutually_exclusive && EmancipationOption.options_with_category_and_case(option_category_id, self[:id]).any?)
+      emancipation_options << EmancipationOption.find(option_id)
+    else
+      raise "Attempted adding multiple options belonging to a mutually exclusive category"
+    end
+  end
```

Comment:
You can simplify this a bit:

```
def add_emancipation_option(option_id)
  option_category = EmancipationOption.find(option_id).emancipation_category

  if !(option_category.mutually_exclusive && EmancipationOption.options_with_category_and_case(option_category.id, id).any?)
    ...
  else
    raise ...
  end
end
```

Note that we don't really write `self[:id]`, but rather `self.id`, and it's okay to leave off `self` since it's implied.